# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../lpeg/hotfix.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- lpeg-0.5/lpeg.c
+++ lpeg-0.5/lpeg.c
@@ -467,7 +467,7 @@
         assert(captop > 0);
         if (capture[captop - 1].siz == 0 &&
             s - capture[captop - 1].s < UCHAR_MAX) {
-          capture[captop - 1].siz = s - capture[captop - 1].s + 1;
+          capture[captop - 1].siz = (s - getoff(p)) - capture[captop - 1].s + 1;
           p++;
           continue;
         }
